prov/efa/src/efa_mr.c

```diff
@@ -142,8 +155,19 @@ static int efa_mr_cache_reg(struct fid *fid, const void *buf, size_t len,
 
 	md = (struct efa_mr *)entry->data;
 	md->entry = entry;
+	attr.requested_key = md->mr_fid.key;
+	shm_attr.requested_key = md->mr_fid.key;
 
 	*mr_fid = &md->mr_fid;
+	if (md->is_new_entry_reg && strstr(domain->util_domain.name, "rdm")) {
```

Comment:
I prefer not distinguish rdm and dgram end point, and do that for both end points.

Reply:
We would still have to differentiate between dgram and rdm, as we dont support shm for dgram.